### PR TITLE
ERC725 Update 

### PR DESCRIPTION
**Pull request by YamenMerhi**
ERC725Alliance Repo Update:
- Changed setData from `calldata` to `memory`
- Took the change of matt regarding the format of the test suite and the names `ERC725YReader/Writer` instead of `Dummy`
- Added revert on failed call + return bytes
- Added helper contract to test ^ these extra features
- Added `universalReceiver` function to ERC725Account and keys needed in constructor
- Added `ERC725Utils` library to make ERC725Account able to set `UniversalReceiverDelegate Key`
- Extended return value of universalReceiver from `bytes32` to `bytes`
- Added Dummy UniversalReceiver to test the function
- Seperated `ERC725X` and `ERC725Y` tests from ERC725Account test
- Fixed the tests to `create2` in ERC725X (was hardcoded without any referance , i implemented `eth-create2-calculator` package)
- Deleted KeyManager and related tests `(ERC734)`
- Calculated `165 interface id`
- Added delagte call
- Added return the address when create1 and create2 
- Clean the test

### CODE REVIEW

**Comment by YamenMerhi** [comment]
@frozeman @CJ42 We should discuss `calldata` to `memory` not just in ERC725Y

File: implementations/contracts/ERC725/ERC725Account.sol [hunk 1]
```diff
@@ -89,4 +101,32 @@ contract ERC725Account is ERC725, IERC1271  {
...
+    {
+        bytes memory receiverData = ERC725Y(this).getDataSingle(_UNIVERSAL_RECEIVER_DELEGATE_KEY);
+        returnValue = "";
+
+        // call external contract
+        if (receiverData.length == 20) {
+            address universalReceiverAddress = BytesLib.toAddress(receiverData, 0);
+            
+            if(ERC165(universalReceiverAddress).supportsInterface(_INTERFACE_ID_LSP1DELEGATE)) {
+                returnValue = ILSP1Delegate(universalReceiverAddress).universalReceiverDelegate(
```
Comment (frozeman) [inline implementations/contracts/ERC725/ERC725Account.sol:120]: Did you test it it fails? On revert inside the ur delegate. Contract ?

File: implementations/contracts/ERC725/IERC725X.sol [hunk 2]
```diff
@@ -48,5 +48,5 @@ interface IERC725X {
         address to,
         uint256 value,
         bytes calldata data
-    ) external payable;
+    ) external payable returns(bytes memory);
```
Comment (frozeman) [inline implementations/contracts/ERC725/IERC725X.sol:51]: Could you also update the ERC725 docs adding the bytes array and return values? Maybe first make the or to this repo before we do at the eip

File: implementations/contracts/ERC725/ERC725XCore.sol [hunk 3]
```diff
@@ -46,15 +46,15 @@ abstract contract ERC725XCore is ERC165Storage, IERC725X {
         address _to,
         uint256 _value,
         bytes calldata _data
```
Comment (frozeman) [inline implementations/contracts/ERC725/ERC725XCore.sol:48]: memory

**Comment by CJ42** [comment]
We aWould be better to rename the folder **ILSP1** to **interfaces**.

File: implementations/contracts/ERC725/ERC725X.sol [hunk 4]
```diff
@@ -36,6 +36,39 @@ contract ERC725X is Ownable, ERC725XCore {
         uint256 _value,
         bytes calldata _data
```
Comment (frozeman) [inline implementations/contracts/ERC725/ERC725X.sol:37]: we wanted to change this as well to memory as we discussed

File: implementations/contracts/ERC725/ERC725XInit.sol [hunk 5]
```diff
@@ -34,6 +34,38 @@ contract ERC725XInit is ERC725XCore, OwnableUpgradeable {
         uint256 _value,
         bytes calldata _data
```
Comment (frozeman) [inline implementations/contracts/ERC725/ERC725XInit.sol:35]: Why do we copy and paste here? we shouldn't use copy and pasted code between normal and init.

**Review by frozeman — CHANGES_REQUESTED** [review]
Please check the comments

**Review by frozeman — APPROVED** [review]
ill merge this and we do the inheritance change in another PR

**Pull request merged**